src/components/GradeLevelFilter.tsx

```diff
@@ -14,26 +15,56 @@ interface GGProps {
   gradeGroup: GradeLevel[];
   grades: Set<GradeLevel>;
   toggleSelection: (s: GradeLevel) => void;
+  onChange: (a: GradeLevel[]) => void;
 }
 
-function GradeGroup({ title, gradeGroup, grades, toggleSelection }: GGProps) {
+function GradeGroup({
+  title,
+  gradeGroup,
+  grades,
+  toggleSelection,
+  onChange,
+}: GGProps) {
+  let allChecked = gradeGroup.every((grade) => grades.has(grade));
+
   return (
-    <FormGroup sx={{ py: 1 }}>
-      <FormLabel>{title}</FormLabel>
-      {gradeGroup.map((grade) => (
+    <Box sx={{ py: 1 }}>
+      <>
         <FormControlLabel
-          key={grade}
+          label={<Typography sx={{ fontWeight: 'bold' }}>{title}</Typography>}
           control={
             <Checkbox
-              color="primary"
-              checked={grades.has(grade)}
-              onChange={() => toggleSelection(grade)}
+              checked={allChecked}
+              onChange={() => {
+                if (!allChecked) {
+                  gradeGroup.forEach((grade) => grades.add(grade));
+                } else {
+                  gradeGroup.forEach((grade) => grades.delete(grade));
+                }
```

Comment:
You could probably do this:
```suggestion
                  gradeGroup.forEach((g) => allChecked ? grades.delete(g) : grades.add(g));
```